packages/config/src/projects/layer2s/optimism.ts

```diff
@@ -436,11 +482,50 @@ export const optimism: Layer2 = {
   },
-  stage: {
-    stage: 'UnderReview',
+  stateValidation: {
+    description:
+      'Updates to the system state can be proposed and challenged by anyone who has sufficient funds. If a state root passes the challenge period, it is optimistically considered correct and made actionable for withdrawals.',
+    categories: [
+      {
+        title: 'State root proposals',
+        description: `Proposers submit state roots as children of the latest confirmed state root (called anchor state), by calling the \`create\` function in the DisputeGameFactory. A state root can have multiple conflicting children. Each proposal requires a stake, currently set to ${formatEther(
+          permissionlessDisputeGameBonds,
+        )} ETH, that can be slashed if the proposal is proven incorrect via a fraud proof. Stakes can be withdrawn only after the proposal has been confirmed. A state root gets confirmed if the challenge period has passed and it is not countered.`,
+        references: [
+          {
+            text: 'OP stack specification: Fault Dispute Game',
+            href: 'https://specs.optimism.io/fault-proof/stage-one/fault-dispute-game.html#fault-dispute-game',
+          },
+        ],
+      },
+      {
+        title: 'Challenges',
+        description: `Challenges are opened to disprove invalid state roots using bisection games. Each bisection move requires a stake that increases expontentially with the depth of the bisection, with a factor of ${exponentialBondsFactor}. The maximum depth is ${permissionlessGameMaxDepth}, and reaching it therefore requires a cumulative stake of ${parseFloat(
+          formatEther(permissionlessGameFullCost),
+        ).toFixed(2)} ETH from depth 0. Actors can participate in any challenge by calling the \`defend\` or \`attack\` functions, depending whether they agree or disagree with the claim and want to move the bisection game forward.
+        Actors might be involved in multiple (sub-)challenges at the same time, meaning that the protocol operates with [full concurrency](https://medium.com/l2beat/fraud-proof-wars-b0cb4d0f452a). Each claim has a clock that starts with ${formatSeconds(
+          maxClockDuration,
+        )}, and that gets passed down to grandchildren claims. If a clock expires, the claim is considered defeated if it was countered, or it gets confirmed if uncountered. Since honest parties can inherit clocks from malicious claims (therefore from malicious parties, see [freeloader claims](https://specs.optimism.io/fault-proof/stage-one/fault-dispute-game.html#freeloader-claims)), if a clock has less than ${formatSeconds(
+          permissionlessGameClockExtension,
+        )}, it generally gets extended by ${formatSeconds(
+          permissionlessGameClockExtension,
+        )} with the exception of ${formatSeconds(
+          permissionlessGameClockExtension * 2,
+        )} at depth ${permissionlessGameSplitDepth}, and ${formatSeconds(
```

Comment:
Should this be 'at depth **>** 30'? I am assuming the clock extension would not exclusively work at depth 30?

Reply:
i thought that too but it's not the case. their spec is a bit confusing and they tweaked the math in the latest upgrade. pls have a look at the FaultDisputeGame contract if you want to confirm

Reply:
wow ok this is complicated. I read the contract and because of this part:
```
if (nextPositionDepth == MAX_GAME_DEPTH - 1) {
            // If the next position is `MAX_GAME_DEPTH - 1` then we're about to execute a step. Our
            // clock extension must therefore account for the LPP challenge period in addition to
            // the standard clock extension.
            actualExtension = CLOCK_EXTENSION.raw() + uint64(VM.oracle().challengePeriod());
        } else if (nextPositionDepth == SPLIT_DEPTH - 1) {
            // If the next position is `SPLIT_DEPTH - 1` then we're about to begin an execution
            // trace bisection and we need to give extra time for the off-chain challenge agent to
            // be able to generate the initial instruction trace on the native FPVM.
            actualExtension = CLOCK_EXTENSION.raw() * 2;
            ```

i think the two depth vars in the text should be both reduced by 1: `at depth ${permissionlessGameSplitDepth-1}` and `at the last-1 depth`